ci/azure/linux.yml

```diff
@@ -155,7 +173,7 @@ jobs:
     - task: PublishPipelineArtifact@1
       inputs:
         path: /tmp/env/env.yml
-        artifact: LinuxCondaEnvironment$(PYTHON_NO_DOT_VERSION)
+        artifact: LinuxCondaEnvironment-$(PIPELINE_ARTIFACT_TAG)
```

Comment:
Why this change too?

Reply:
The `LinuxTest py3X_pyspark_spark` jobs would fail on this step because they are trying to publish an artifact with the same name as an existing artifact

Our existing jobs already do publish artifacts with duplicate names completely fine, but I'm guessing the error comes up specifically for the `_pyspark_spark` jobs because the artifact is different for those jobs (contents of `/tmp/env/env.yml` is different, see line 175, line 135). So the jobs are trying to publish that different artifact to an existing name

E.g., this is what would happen before this change:

1. `LinuxTest py36_omnisci` publishes artifact `LinuxCondaEnvironment36`
2. `LinuxTest py36_sql_parquet` publishes artifact `LinuxCondaEnvironment36` (no error, possibly because the contents of the artifact input `/tmp/env/env.yml` is the same as above?)
3. `LinuxTest py36_pyspark_spark` tries to publish artifact `LinuxCondaEnvironment36` (with an artifact input `/tmp/env/env.yml` that has different contents) and fails with this error: `Artifact LinuxCondaEnvironment36 already exists for build 2515.`